src/actions/actions.ts

```diff
@@ -121,13 +121,15 @@ export class ActionPayloadTypes {
      * @param ignorePageLimit Page size should be ignored
      * @param keepDelta Pending changes should not be dropped when performing this request (due to
      * hierarchy expanging through cursor change, for same BC hierarchy this leads to data loss)
+     * @param paginationMode From first to current page
      */
     bcFetchDataRequest: {
         bcName: string,
         depth?: number,
         widgetName: string,
         ignorePageLimit?: boolean,
-        keepDelta?: boolean
+        keepDelta?: boolean,
+        paginationMode?: boolean
```

Comment:
1) We should try to keep actions simple and directly connected to user interaction.  
Different use cases equals different actions, as your can see `bcFetchDataRequest` action is already overloaded with totally unintuitive parameters `depth`, `ignorePageLimit` and `keepDelta`, and they'll keep piling up.
I think we need to introduce a new action instead, something like:
```tsx
/**
 * Fetch data request request for specific pages range
 */
bcFetchDataPages: {
    /**
     * The business component to fetch data for
     * @deprecated TODO: 2.0.0 Should be removed in favor of widgetName
     */
    bcName: string,
    /**
     * Fisrt page to fetch (default is 1)
     */
    from?: number,
    /**
     * Last page to fetch (default is current page)
     */
    to?: number
} = z
```
then add this action to epic
```tsx
const bcFetchDataEpic: Epic = (action$, store) => action$.ofType(
    types.bcFetchDataRequest,
    types.bcFetchDataPages,
    // ...other actions
)
```
then 
```tsx
const fetchParams: Record<string, any> = {
    _page: page,
    _limit: limit
}
if (action.type === types.bcFetchDataPages) {
    fetchParams._page = action.payload.from || 1
    fetchParams._limit = (action.payload.to || page - fetchParams._page) * limit
}
```

2) `paginationMode` property name suggests that it'll be either enum or union of some string constants.  
It is difficult to grasp what exactly will happen if this option is active without looking into implementation. So the property shouldn't be boolean or should be named like "allCurrentPages`

Reply:
resolved in another PR (#102)